taichi/rhi/vulkan/vulkan_device.cpp

```diff
@@ -1977,10 +1977,6 @@ DeviceAllocation VulkanDevice::create_image(const ImageParams &params) {
         vkapi::create_image_view(device_, alloc.image, &view_info));
   }
 
-  if (params.initial_layout != ImageLayout::undefined) {
-    image_transition(handle, ImageLayout::undefined, params.initial_layout);
```

Comment:
Why do we skip this?

Reply:
Although the Vulkan API always create image in undefined layout, but in our RHI we allow setting an initial layout to make it easier for the programmer